src/hexo/themes/documentation/source/js/scanner.js

```diff
@@ -0,0 +1,312 @@
+        responseItems.forEach(function (responseItem) {
+            if (!responseItem.results) {
+                return responseItem.results;
+            }
+
+            responseItem.results = responseItem.results.filter(function (result) {
+                return !(existingResults[responseItem.name] && existingResults[responseItem.name].includes(result.name));
+            });
+        });
+
+        return responseItems;
+    };
+
+    var generateTemplate = function () {
+        var tmpl = `{{#each categories}}
+        {{#if results}}
+        <section class="rule-result" id="{{name}}">
+            <h3>{{name}}</h3>
+            {{#each results}}
+            <div class="rule-result--details" aria-expanded="false">
+                <div class="rule-result--details__header">
+                    <p class="rule-title">{{name}}: {{getLength messages status}}</p>
+                    <div class="rule-result__docs">
+                        <a href="https://sonarwhal.com/docs/user-guide/rules/{{name}}.html"><img src="/images/results-docs-icon.svg" alt="documentation"  class="docs-icon" /></a>
+                        <button title="show warning details" class="button--details">Close Details</button>
+                    </div>
+                </div>
+                {{#each messages}}
+                <div class="rule-result--details__body">
+                    <p class="warning-badge uppercase-text">{{../status}}</p>
+                    <p>
+                        {{message}}
+                    </p>
+                    <div class="rule-result__code">
+                        <p>
+                            {{cutUrlString resource}}
+                            {{#if location.line}}:{{location.line}}{{/if}}
+                            {{#if location.column}}:{{location.column}}{{/if}}
+                        </p>
+                        <code>{{cutCodeString sourceCode}}</code>
+                    </div>
+                </div>
+                {{/each}}
+            </div>
+            {{/each}}
+        </section>
+        {{/if}}
+    {{/each}}`;
+
+        return tmpl;
+    };
+    var cutString = (string, lengthToShow) => {
+        if (!string || string.length < lengthToShow) {
+            return string;
+        }
+
+        return `${string.slice(0, lengthToShow)} ... ${string.slice(string.length - lengthToShow)}`;
+    };
+
+    var registerToggleExpandListener = function () {
+        var detailButtons = arraify(document.querySelectorAll('.button--details'));
+
+        detailButtons.map(function (button) {
+            button.addEventListener('click', toggleExpand, false);
+        });
+    };
+
+<<<<<<< HEAD
+=======
+    var closeOverlay = function () {
+        document.querySelector('.overlay').classList.remove('open');
+    };
+
+    var openOverlay = function () {
+        document.querySelector('.overlay').classList.add('open');
+    };
+
+>>>>>>> 01e4ebed1a0d513f52c3bc5f5c9214635cbb29cc
```

Comment:
@qzhou1607 there's a conflict in this file. Can you update this before I do a pull?